homeassistant/components/remote_rpi_gpio/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Support for binary sensor using RPi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import remote_rpi_gpio
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDRESS = 'address'
+CONF_BOUNCETIME = 'bouncetime'
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_PORTS = 'ports'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_BOUNCETIME = 50
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_PULL_MODE = "UP"
+
+DEPENDENCIES = ['remote_rpi_gpio']
+
+_SENSORS_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
Please move the config to the component and set up the platforms via discovery. See octoprint component for example.

Reply:
I envision the platform being utilized under the binary_sensor or switch platform and not necessarily set up under it's own platform (as shown in my config example).  It seem that if it is done this way I can't up the platform via discovery.  See rpi_gpio as an example.

ps
I did change it to set up the platform via discovery as suggested and tested with a remote_rpi_gpio: section in my config and it worked, but when I removed that and replaced it with the config shown in my example entry it didn't work.

Reply:
See the netgear_lte component for example. The component config will have options for the platforms under the corresponding platform keys, eg `'switch'`.

Please make a try at it, then commit and push it. Then I can help you get it correct.